acquire/collector.py

```diff
@@ -289,114 +285,364 @@ def collect(
+                # recursively calling _get_symlink_branches(), we only need to walk
+                # over the remaining parts to see if any of them are symlinks.
+                cur_path = cur_path.resolve()
+
+        last_part = path.parts[-1]
+        path = cur_path.joinpath(last_part)
+
+        return path, branches
+
+    def collect_path(
         self,
-        path: Union[str, fsutil.TargetPath],
-        size: Optional[int] = None,
+        path: str | fsutil.TargetPath,
         outpath: Optional[str] = None,
         module_name: Optional[str] = None,
         base: Optional[str] = None,
         volatile: bool = False,
+        seen_paths: set[fsutil.TargetPath] = None,
     ) -> None:
+        """Collect a path from the target's root filesystem, including any intermediary symlinks.
+
+        Args:
+            path: The path to collect (this may be a file, directory or symlink).
+            outpath: A posix style explicit path where to store the collected path. In case ``path``
+                     is a directory this will be the new base directory. It is concatenated with
+                     ``base`` to get the final output path. Windows device path and sysvol parts are
+                     normalized. When set, intermediate symlinks of ``path`` are not collected. When
+                     not set, it will be constructed from the given ``path``.
+            module_name: When set it indicates the module doing the collection, used for logging and
+                         reporting. When not set the ``Collector``'s ``bound_module`` will be used.
+            base: A different base path to use to store the file, it is prepended to the given or
+                  generated ``outpath``.
+            volatile: When this flag is set, the collection of a number of artefacts is done
+                      slightly different:
+                      - symlinks at the end of a path will not be collected,
+                      - empty directories will be collected,
+                      - files will be collected in a slower but more robust way, any errors while
+                        reading the bytes will not fail the collection of the file and all bytes
+                        already retrieved will be stored.
+            seen_paths: A list of normalized path strings, used when calling this function
+                        recursively to collect directories to break out of symlink loops.
+        """
         module_name = self.bound_module_name or module_name
         if not module_name:
             raise ValueError("Module name must be provided or Collector needs to be bound to a module")
 
         if not isinstance(path, fsutil.TargetPath):
             path = self.target.fs.path(path)
 
-        if self.filter(path) is True:
-            log.info("- Collecting file %s: Skipped (filtered out)", path)
-            return
+        log.info("- Collecting path %s", path)
```

Comment:
With the new logging I get a lot of seemingly duplicate log lines printed to the console, can that be cleaned up more? Maybe log some things to debug (the log file) and only print the minimal amount of log lines necessary to the console (info).

For example, this is what I see for every single file:
```
- Collecting path /c:/Users/User/AppData/Local/Microsoft/Windows/UsrClass.dat.LOG1
- Collecting file /c:/Users/User/AppData/Local/Microsoft/Windows/UsrClass.dat.LOG1 to: fs/c:/Users/User/AppData/Local/Microsoft/Windows/UsrClass.dat.LOG1
- Collecting file /c:/Users/User/AppData/Local/Microsoft/Windows/UsrClass.dat.LOG1 succeeded
- Collecting path /c:/Users/User/AppData/Local/Microsoft/Windows/UsrClass.dat.LOG1 succeeded
```

I liked the `... : OK` that we previously had, but I also like how now you're more easily able to track progress. Maybe something like this?
```
- Collecting file /c:/Users/User/AppData/Local/Microsoft/Windows/UsrClass.dat.LOG1 ...
- ... Done!
```

Maybe we can move towards `structlog` in the future to better facilitate this too.

Reply:
Yeah, that's a bit unfortunate. The initial collection will always be a 'path' and it is not clear upfront if this will result in a single file, directory or something else with or without a set of intermediate symlinks.

I don't really see a different way to indicate which set of artifacts belong to a certain path, while retaining the traceability where and which things fail to collect.